File: packages/lector/test/fixtures/go-reference/app/stripe.go
package app

import "fixture.lector.invalid/gomod-reference/contracts"

type StripeProcessor struct{}

func (StripeProcessor) Process(order contracts.Order) (contracts.Receipt, error) {
	return contracts.Receipt{Order: order, Processed: true}, nil
}

func CreateProcessor() contracts.PaymentProcessor {
	return StripeProcessor{}
}

File: packages/lector/test/fixtures/rust-reference/app/src/stripe.rs
use contracts::{Order, PaymentProcessor, Receipt};

pub struct StripeProcessor;

impl PaymentProcessor for StripeProcessor {
    fn process(&self, order: Order) -> Receipt {
        Receipt { order, processed: true }
    }
}

pub fn create_processor() -> impl PaymentProcessor {
    StripeProcessor
}
